Source/Krypton Toolkit/Krypton.Toolkit.Suite.Extended.DataGridView/Krypton.Toolkit.Suite.Extended.DataGridView 2019.csproj

```diff
@@ -40,27 +40,23 @@
         <OutputPath>..\..\..\..\Bin\$(configuration)\Main\Krypton.Toolkit.Suite.Extended.DataGridView\</OutputPath>
     </PropertyGroup>
   
-  <!--TODO: Please fix, as this causes the compiler to go crazy-->
-  <ItemGroup>
-    <!--<PackageReference Include="System.ValueTuple" Version="4.5.0" />-->
-  </ItemGroup>
-
-  <!--<ItemGroup>
-    <Reference Include="System.ValueTuple, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="$(TargetFramework.StartsWith('net4'))">
-      <SpecificVersion>True</SpecificVersion>
-      <Version>4.0.0.0</Version>
-      <TreatAsUsed>True</TreatAsUsed>
-    </Reference>
-    <Reference Include="System.ValueTuple, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="$(TargetFramework.StartsWith('net5'))">
-      <SpecificVersion>True</SpecificVersion>
-      <Version>5.0.0.0</Version>
-      <TreatAsUsed>True</TreatAsUsed>
-    </Reference>
-    <Reference Include="System.ValueTuple, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="$(TargetFramework.StartsWith('net6'))">
-      <SpecificVersion>True</SpecificVersion>
-      <Version>6.0.0.0</Version>
-      <TreatAsUsed>True</TreatAsUsed>
-    </Reference>
-    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net35'" />
-  </ItemGroup>-->
+  <!--TODO: Help!!-->
+  <Choose>
+    <When Condition="'$(TargetFramework)' <= 'net4'">
+      <ItemGroup>
+        <Reference Include="System.ValueTuple, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="$(TargetFramework.StartsWith('net4'))">
+          <SpecificVersion>True</SpecificVersion>
+          <Version>4.0.0.0</Version>
+          <TreatAsUsed>True</TreatAsUsed>
+        </Reference>
+        <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net35'" />
```

Comment:
This is in the wrong place, it should be in it's own condition
or remove the Choose when clauses and just have them sort themselves out like in the `System.Designer` styles

Reply:
@Smurf-IV Do you mean like so?

```
<Choose>
    <When Condition="'$(TargetFramework.StartsWith('net4'))">
      <ItemGroup>
        <Reference Include="System.ValueTuple, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
          <SpecificVersion>True</SpecificVersion>
          <Version>4.0.0.0</Version>
          <TreatAsUsed>True</TreatAsUsed>
        </Reference>
      </ItemGroup>
    </When>
    <Otherwise>
      <ItemGroup>
        <PackageReference Include="System.ValueTuple" Version="4.5.0" />
      </ItemGroup>
    </Otherwise>
  </Choose>

  <ItemGroup>
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net35'" />
  </ItemGroup>
```

Reply:
No, Just like the last line with the conditions for the other frameworks.
Bin the `Choose` / `When`/ conditions because they are unreliable for references.
And are you sure it needs to be a `PackageReference` for .net35 and core 3 / core 5 etc.?

Reply:
Ok updated it to:

```
<ItemGroup>
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net35'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net40'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net45'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net451'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net452'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net46'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net461'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net462'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net47'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net471'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net472'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net48'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'netcoreapp3.1'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net5'" />
    <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net6'" />
  </ItemGroup>
```

Reply:
So it's the same reference regardless of the TFM.
So why the need for the individual ones, as they are not explicitly asking for a version !

Reply:
I'm confused, it seems that this part is the major problem blocking the build, as `<PackageReference Include="System.ValueTuple" Version="4.5.0" />` works only for .NET Core and up. The other bugs are easy to fix.

Reply:
> 
> 
> Ok updated it to:
> 
> ```
> <ItemGroup>
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net35'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net40'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net45'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net451'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net452'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net46'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net461'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net462'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net47'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net471'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net472'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net48'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'netcoreapp3.1'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net5'" />
>     <Reference Include="System.ValueTuple" Condition="'$(TargetFramework)' == 'net6'" />
>   </ItemGroup>
> ```

If the above works, without any `PackageReference`, then the following will work
`<Reference Include="System.ValueTuple" />

Reply:
According to the build log, it hangs on `Extended-Toolkit\Source\Krypton Toolkit\Krypton.Toolkit.Suite.Extended.DataGridView\Implementation\MasterDetailGridViewBase.cs(247,44): error CS8179: Predefined type 'System.ValueTuple`2' is not defined or imported`

Reply:
On What TFM Is that for ?

Reply:
4.8 & lower